lib/entice/logic/casting.ex

```diff
@@ -8,51 +8,115 @@ defmodule Entice.Logic.Casting do
   alias Entice.Logic.Casting
   alias Entice.Logic.Vitals.Energy
   alias Entice.Entity
-  alias Entice.Entity.Coordination
+  use Pipe
 
 
   defstruct(
-    casting_timer: nil,
+    casting_timer: nil,  #Why not call this cast_timer if skill has a cast_time ? or vice versa
     after_cast_timer: nil,
     recharge_timers: %{})
 
-
   def register(entity_id),
-  do: Entity.put_behaviour(entity_id, Casting.Behaviour, [])
-
+  do: Entity.put_behaviour(entity_id, Casting.Behaviour, %Casting{})
```

Comment:
thats such a great idea to initialize the behaviour with the attribute we want to add :D
i like that... maybe we should do that with other stuff as well some time in the future
also enables us to initialize the entities in a more simple fashion, e.g. for testing